src/wallet.rs

```diff
@@ -296,26 +313,36 @@ pub fn parse_command(
 }
 
 pub fn process_command(config: &WalletConfig) -> Result<String, Box<error::Error>> {
+    if let WalletCommand::Address = config.command {
+        // Get address of this client
+        return Ok(format!("{}", config.id.pubkey()));
+    }
+
+    let leader = poll_gossip_for_leader(config.network, config.timeout)?;
+    let tpu_addr = leader.contact_info.tpu;
+    let drone_addr = config.drone_addr(tpu_addr);
+    let rpc_addr = config.rpc_addr(tpu_addr);
```

Comment:
This can be updated to `let rpc_addr = leader.contact_info.rpc;`
So we don’t need to pass `rpc_port` through anymore. Still need to check `proxy`, though.

Reply:
But `proxy` still overrides `leader.contact_info.rpc`?

Reply:
Yep